### PR TITLE
New UI-Kit Based theme and Paragraphs Page Builder

### PR DESCRIPTION
**Pull request by IvanZugec**
This PR contains the work done on a new UI-Kit starter theme and new Paragraph types for building pages.

### CODE REVIEW

**Comment by pandaskii** [comment]
Behat testing may need an update to adopt the changes

**Comment by tobybellwood** [comment]
yup - there's a lot to cover here!

In summary:

**modules/custom/core/govcms_core**
	A lot of the config for paragraphs, image styles is in here
**modules/custom/core/govcms_media**
	Core media implementation, with media browser, multiple file upload, and media library tagging
**modules/custom/core/govcms_search**
	Initial search configuration
**modules/custom/core/govcms_security**
	Placeholder for GovCMS security config
**modules/custom/core/govcms8_calendar_item**
	Field formatter for date fields - to be used with base theme
**modules/custom/core/govcms8_default_content**
	Example content to populate GovCMS site - to be used with base theme
**modules/custom/core/govcms8_layout**
	The page/view twig templates - to be used with base theme
**modules/custom/core/govcms8_modifiers**
	The modifiers config - to be used with base theme

**modules/custom/fields/govcms_fields_attachment**
	Core media implementation for Attachment field type
**modules/custom/fields/govcms_fields_featured_image**
	Core media implementation for Featured Image field type

**modules/custom/optional/govcms_optional_govcms_blog_article**
	Default content type, optimized for base theme, utilising core media
**modules/custom/optional/govcms_optional_govcms_event**
	Default content type, optimized for base theme, utilising core media
**modules/custom/optional/govcms_optional_govcms_foi**
	Default content type, optimized for base theme, utilising core media
**modules/custom/optional/govcms_optional_govcms_news_and_media**
	Default content type, optimized for base theme, utilising core media
**modules/custom/optional/govcms_optional_govcms_standard_page**
	Default content type, optimized for base theme, utilising core media

**themes/custom/govcms8_install**
	Custom theme to provide installation screens
**themes/custom/govcms8_uikit**
	Twig templates to convert UIKIT elements for use with base theme
**themes/custom/govcms8_uikit_starter**
	Base GovCMS8 theme, built using UIKIT, featuring built-in KSS styleguide

File: modules/custom/core/govcms8_default_content/govcms8_default_content.info.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+name: GovCMS 8 Default Content
+type: module
+description: Provides default content for G8BT.
```
Comment (tobybellwood) [inline modules/custom/core/govcms8_default_content/govcms8_default_content.info.yml]: Would placing this module under modules/custom/optional be a better place for it (at the moment) to offer people the option to install it up front (alongside the content types)?

Regardless - we should probably tweak the description here to give people a better idea of what it does.  We also need to ensure that it resolves dependencies on the predefined content types.

"Provides example content and structure to build a sample OOTB site, complete with a Styleguide and a 'kitchen sink' page"

Comment (tobybellwood) [inline modules/custom/core/govcms8_default_content/govcms8_default_content.info.yml]: Would placing this module under modules/custom/optional be a better place for it (at the moment) to offer people the option to install it up front (alongside the content types)?We

File: govcms.info.yml [hunk 2]
```diff
@@ -60,6 +65,12 @@ dependencies:
   # govCMS core modules
   - govcms_search
   - govcms_media
+  - govcms_core
+  - govcms_optional_govcms_blog_article
```
Comment (IvanZugec) [inline govcms.info.yml]: @tobybellwood should I remove the optional modules from govcms.info.yml?

File: modules/custom/core/govcms8_default_content/govcms8_default_content.info.yml (hunk 1 above)
Comment (IvanZugec) [reply modules/custom/core/govcms8_default_content/govcms8_default_content.info.yml]: Okay, will do.

File: themes/custom/govcms8_uikit_starter/styleguide/index.html [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+</head>
+<body id="kss-node" >
+
+<div class="kss-sidebar au-body au-body--dark">
+  <header class="kss-header">
+    <h1 class="kss-doc-title">GovCMS 8 UI-Kit Starter Style Guide</h1>
+  </header>
+  <nav class="kss-nav">
+    <ul class="kss-nav__menu">
```
Comment (tobybellwood) [inline themes/custom/govcms8_uikit_starter/styleguide/index.html]: This menu needs a link to take you back to the main site.

Is it ok to use "../../../../../../"?

File: themes/custom/govcms8_uikit_starter/styleguide/item-ui-kit.html [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+
+</head>
+<body id="kss-node" class="kss-fullscreen-mode">
+
+<div class="kss-sidebar au-body au-body--dark">
+  <header class="kss-header">
+    <h1 class="kss-doc-title">GovCMS 8 UI-Kit Starter Style Guide</h1>
+  </header>
+  <nav class="kss-nav">
+    <ul class="kss-nav__menu">
```
Comment (tobybellwood) [inline themes/custom/govcms8_uikit_starter/styleguide/item-ui-kit.html]: This menu needs a link to take you back to the main site.

Is it ok to use "../../../../../../"?

File: modules/custom/optional/govcms8_default_content/govcms8_default_content.info.yml [hunk 5]
```diff
@@ -0,0 +1,13 @@
+name: GovCMS 8 Default Content
```
Comment (tobybellwood) [inline modules/custom/optional/govcms8_default_content/govcms8_default_content.info.yml:1]: I'll organise the service design/comms team to go over the text used for the default content

File: modules/custom/core/govcms8_modifiers/config/install/field.storage.paragraph.field_mod_image.yml [hunk 6]
```diff
@@ -0,0 +1,20 @@
+uuid: 75cd7c3a-165a-4ab6-972e-6bf9c3779a79
+langcode: en
+status: true
+dependencies:
+  module:
+    - media
+    - paragraphs
+id: paragraph.field_mod_image
```
Comment (tobybellwood) [inline modules/custom/core/govcms8_modifiers/config/install/field.storage.paragraph.field_mod_image.yml:8]: Can we make the Image Background Modifier use the same media browser as the featured image field for consistency?

File: modules/custom/core/govcms8_modifiers/config/install/paragraphs.paragraphs_type.parallax_bg_modifier.yml [hunk 7]
```diff
@@ -0,0 +1,9 @@
+uuid: f95f4411-a65d-4bd5-949e-356ec7f4a276
+langcode: en
+status: true
+dependencies: {  }
+id: parallax_bg_modifier
+label: 'Parallax Background Modifier'
```
Comment (tobybellwood) [inline modules/custom/core/govcms8_modifiers/config/install/paragraphs.paragraphs_type.parallax_bg_modifier.yml:6]: Can we make the Parallax Background Modifier use the same media browser as the featured image field?

Comment (IvanZugec) [reply modules/custom/core/govcms8_modifiers/config/install/paragraphs.paragraphs_type.parallax_bg_modifier.yml:6]: Fixed in e2cb3b8cb13a28334ef135040adb76017c0a2098

File: modules/custom/core/govcms8_modifiers/config/install/field.storage.paragraph.field_mod_image.yml (hunk 6 above)
Comment (IvanZugec) [reply modules/custom/core/govcms8_modifiers/config/install/field.storage.paragraph.field_mod_image.yml:8]: Fixed in e2cb3b8cb13a28334ef135040adb76017c0a2098

File: themes/custom/govcms8_uikit_starter/styleguide/index.html (hunk 3 above)
Comment (IvanZugec) [reply themes/custom/govcms8_uikit_starter/styleguide/index.html]: Added a link "< Back to site" in c0f235a43446bde9dc5f3cee5692a10743d25a49

File: themes/custom/govcms8_uikit_starter/styleguide/item-ui-kit.html (hunk 4 above)
Comment (IvanZugec) [reply themes/custom/govcms8_uikit_starter/styleguide/item-ui-kit.html]: Added a link "< Back to site" in c0f235a43446bde9dc5f3cee5692a10743d25a49

**Comment by tobybellwood** [comment]
Module structure overview for govcms8 theme modules

(is there a better name for govcms_core - ppl may confuse it with the core govcms8 config)

* govcms_core
    + govcms8_modifiers
        - paragraphs
        - modifiers
        - entity_class_formatter
        - entity_reference_display
        - (jarallax library)
    + govcms8_layout
        - layout_discovery
    + govcms8_calendar_item

Under a composer world, could the top level module theoretically be the one that's required, and it includes the other three as submodules - this might make it less complex?

**Comment by IvanZugec** [comment]
@tobybellwood Removed the govcms8_uikit_* themes.

**Comment by IvanZugec** [comment]
@tobybellwood I've looked at other distros and they all seem to have a `_core` module:
- varbase_core
- lightning_core
- social_core (open social)

Honestly, I can't think of another module name right now.

File: modules/custom/optional/govcms8_default_content/govcms8_default_content.install [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+ * Implements hook_install().
+ */
+function govcms8_default_content_install() {
+  if (!\Drupal::service('config.installer')->isSyncing()) {
+    \Drupal::classResolver()->getInstanceFromDefinition(InstallHelper::class)->importContent();
+
+    $menu_link = MenuLinkContent::create([
+      'title' => 'Styleguide',
+      'link' => [
+        'uri' => 'internal:/profiles/govcms/themes/custom/govcms8_uikit_starter/styleguide/'
```
Comment (tobybellwood) [inline modules/custom/optional/govcms8_default_content/govcms8_default_content.install]: 'uri' => 'internal:/themes/contrib/govcms8_uikit_starter/styleguide/' now that the theme is a contrib module

**Comment by IvanZugec** [comment]
@tobybellwood fixed uri to styleguide. Good catch.

**Pull request merged**